src/Halogen/Query/HalogenM.purs

```diff
@@ -123,10 +130,20 @@ checkSlot p = HalogenM $ liftF $ CheckSlot p identity
 getRef :: forall s f g p o m. RefLabel -> HalogenM s f g p o m (Maybe Element)
 getRef p = HalogenM $ liftF $ GetRef p identity
 
--- | Provides a way of having a component subscribe to an `EventSource` from
--- | within an `Eval` function.
-subscribe :: forall s f g p o m. ES.EventSource f m -> HalogenM s f g p o m Unit
-subscribe es = HalogenM $ liftF $ Subscribe es unit
+-- | Subscribes a component to an `EventSource`.
+subscribe :: forall s f g p o m. ES.EventSource m f -> HalogenM s f g p o m SubscriptionId
+subscribe es = HalogenM $ liftF $ Subscribe (\_ -> es) identity
+
+-- | An alternative to `subscribe`, intended for subscriptions that unsubscribe
+-- | themselves. Instead of returning the `SubscriptionId` from `subscribe'`, it
+-- | is passed into an `EventSource` constructor. This allows emitted queries
+-- | to include the `SubscriptionId`, rather than storing it in the state of the
+-- | component.
+subscribe' :: forall s f g p o m. (SubscriptionId -> ES.EventSource m f) -> HalogenM s f g p o m Unit
+subscribe' esc = HalogenM $ liftF $ Subscribe esc (const unit)
```

Comment:
As I understand `subscribe'` is same as `void $ subscribe` and if it's correct, that why do we need this?
it's easy to add `void $ ` or `_ <-` in front of `subscribe`

Reply:
It's not the same as `subscribe`:
``` purescript
subscribe :: forall s f g p o m. ES.EventSource m f -> HalogenM s f g p o m SubscriptionId
subscribe' :: forall s f g p o m. (SubscriptionId -> ES.EventSource m f) -> HalogenM s f g p o m Unit
```

Reply:
Just looked at the result 🤦‍♂️